packages/bundler-plugin-core/src/index.ts

```diff
@@ -121,53 +118,87 @@ export function sentryUnpluginFactory({
       });
-
       plugins.push(releaseInjectionPlugin(injectionCode));
     }
 
-    if (options.sourcemaps?.assets) {
-      plugins.push(debugIdInjectionPlugin());
-    }
-
-    if (releaseName) {
+    if (!options.release.name) {
+      logger.warn("No release name provided. Will not manage release.");
+    } else if (!options.authToken) {
+      logger.warn("No auth token provided. Will not manage release.");
+    } else if (!options.org) {
+      logger.warn("No org provided. Will not manage release.");
+    } else if (!options.project) {
+      logger.warn("No project provided. Will not manage release.");
+    } else {
       plugins.push(
         releaseManagementPlugin({
           logger,
-          cliInstance: cli,
-          releaseName: releaseName,
-          shouldCleanArtifacts: options.cleanArtifacts,
-          shouldUploadSourceMaps: options.uploadSourceMaps,
-          shouldFinalizeRelease: options.finalize,
-          include: options.include,
-          setCommitsOption: options.setCommits,
-          deployOptions: options.deploy,
-          dist: options.dist,
+          releaseName: options.release.name,
+          shouldCreateRelease: options.release.create,
+          shouldCleanArtifacts: options.release.cleanArtifacts,
+          shouldFinalizeRelease: options.release.finalize,
+          include: options.release.uploadLegacySourcemaps,
+          setCommitsOption: options.release.setCommits,
+          deployOptions: options.release.deploy,
+          dist: options.release.dist,
           handleRecoverableError: handleRecoverableError,
           sentryHub,
           sentryClient,
+          sentryCliOptions: {
+            authToken: options.authToken,
+            org: options.org,
+            project: options.project,
+            silent: options.silent,
+            url: options.url,
+            vcsRemote: options.release.vcsRemote,
+            headers: options.headers,
+          },
         })
       );
     }
 
-    if (!unpluginMetaContext.watchMode && options.sourcemaps?.assets !== undefined) {
-      plugins.push(
-        debugIdUploadPlugin({
-          assets: options.sourcemaps.assets,
-          ignore: options.sourcemaps.ignore,
-          dist: options.dist,
-          releaseName: releaseName,
-          logger: logger,
-          cliInstance: cli,
-          handleRecoverableError: handleRecoverableError,
-          sentryHub,
-          sentryClient,
-        })
-      );
+    if (options.sourcemaps) {
+      if (!options.authToken) {
+        logger.warn("No auth token provided. Will not upload source maps.");
+      } else if (!options.org) {
+        logger.warn("No org provided. Will not upload source maps.");
+      } else if (!options.project) {
+        logger.warn("No project provided. Will not upload source maps.");
+      } else if (!options.sourcemaps.assets) {
+        logger.warn("No assets defined. Will not upload source maps.");
+      } else {
```

Comment:
l: Personally I'd prefer to extract this validation to a function (per plugin) to declutter this one a little but feel free to leave it as is as.

Reply:
This currently has a use for TS. We could put it into a helper but I am not sure if it will actually simplify the code by a lot. Gonna leave this for now.